src/components/ec-button-group/ec-button-group.spec.js

```diff
@@ -0,0 +1,218 @@
+      withMockedConsole((errorSpy) => {
+        mountButtonGroup({
+          items: [
+            { value: 'yes' },
+            { text: 'Maybe' },
+          ],
+        });
+        expect(errorSpy).toHaveBeenCalled();
+        expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: custom validator check failed for prop "items"');
+      });
+    });
+  });
+
+  describe(':modal', () => {
+    it('should render the button group and change the value when you click on the current not selected', () => {
+      const wrapper = mountButtonGroupAsTemplate(
+        '<ec-button-group v-model="value" :items="items"></ec-button-group>',
+        {},
+        {
+          data() {
+            return {
+              value: 'yes',
+              items: [
+                { text: 'Yes', value: 'yes' },
+                { text: 'No', value: 'no' },
+              ],
+            };
+          },
+        },
+      );
+      expect(wrapper.vm.value).toBe('yes');
+      expect(wrapper.find('.ec-btn--outline').text()).toBe('No');
+      wrapper.find('.ec-btn--outline').trigger('click');
+      expect(wrapper.find('.ec-btn--outline').text()).toBe('Yes');
+      expect(wrapper.vm.value).toBe('no');
+    });
+
+    it('should render the button group without any selected and change the value when you click on the current not selected', () => {
+      const wrapper = mountButtonGroupAsTemplate(
+        '<ec-button-group v-model="value" :items="items"></ec-button-group>',
+        {},
+        {
+          data() {
+            return {
+              value: '',
+              items: [
+                { text: 'Yes', value: 'yes' },
+                { text: 'No', value: 'no' },
+              ],
+            };
+          },
+        },
+      );
+      expect(wrapper.vm.value).toBe('');
+      wrapper.find('.ec-btn--outline').trigger('click');
+      expect(wrapper.find('.ec-btn--outline').text()).toBe('No');
+      expect(wrapper.vm.value).toBe('yes');
+    });
+
+    it('should render the button group and not change when click on the disabled item', () => {
+      const wrapper = mountButtonGroupAsTemplate(
+        '<ec-button-group v-model="value" :items="items"></ec-button-group>',
+        {},
+        {
+          data() {
+            return {
+              value: 'yes',
+              items: [
+                { text: 'Yes', value: 'yes' },
+                { text: 'No', value: 'no', disabled: true },
+              ],
+            };
+          },
+        },
+      );
+      expect(wrapper.vm.value).toBe('yes');
+      expect(wrapper.find('.ec-btn--outline').text()).toBe('No');
+      wrapper.find('.ec-btn--outline').trigger('click');
+      expect(wrapper.find('.ec-btn--outline').text()).toBe('No');
+      expect(wrapper.vm.value).toBe('yes');
+    });
+
+    it('should render the button group and change the model when click on the button with a number values', () => {
```

Comment:
I find this somewhat redundant, values should be anything really , as suggested by @mcibique

Reply:
I suggested to remove the types checking and these tests are redundant then. The button group always do $emit('change', button.value).

Reply:
Done!